federation/pkg/dnsprovider/providers/google/clouddns/internal/interfaces/interfaces.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package interfaces
+
+import (
+	"google.golang.org/api/googleapi"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/rrstype"
+)
+
+// Interfaces to directly mirror the Google Cloud DNS API structures.
+// See https://godoc.org/google.golang.org/api/dns/v1 for details
+// This facilitates stubbing out Google Cloud DNS for unit testing.
+// Only the parts of the API that we use are included.
+// Others can be added as needed.
+
+type (
+	Change interface {
+		Additions() []ResourceRecordSet
+		Deletions() []ResourceRecordSet
+		// Id() string  // TODO: Add as needed
```

Comment:
There seems to be quite a bit of commented out code in this file. Please remove all of them and retain a single TODO per interface listing the methods. Something like:

```
// TODO: Add Id(), Kind(), StartTime(), Status() as needed. Please see .../cloudprovider/... for reference.
```